policy/modules/contrib/wireguard.te

```diff
@@ -0,0 +1,44 @@
+policy_module(wireguard, 1.0.0)
+
+########################################
+#
+# Declarations
+#
+
+attribute_role wg_roles;
+roleattribute system_r wg_roles;
+
+type wg_quick_t;
+type wg_quick_exec_t;
+init_daemon_domain(wg_quick_t, wg_quick_exec_t)
+role wg_roles types wg_quick_t;
+
+type wg_quick_unit_file_t;
+systemd_unit_file(wg_quick_unit_file_t)
+
+permissive wg_quick_t;
+
+########################################
+#
+# wireguard local policy
+#
+allow wg_quick_t self:fifo_file rw_fifo_file_perms;
+allow wg_quick_t self:netlink_generic_socket create_socket_perms;
+allow wg_quick_t self:netlink_route_socket rw_netlink_socket_perms;
+allow wg_quick_t self:unix_dgram_socket create_socket_perms;
+allow wg_quick_t self:unix_stream_socket create_stream_socket_perms;
+
+kernel_request_load_module(wg_quick_t)
+
+corecmd_exec_bin(wg_quick_t)
+
+files_read_etc_files(wg_quick_t)
+
+auth_read_passwd(wg_quick_t)
+
+miscfiles_read_localization(wg_quick_t)
+
+sysnet_run_ifconfig(wg_quick_t, wg_roles)
+
+domain_use_interactive_fds(wg_quick_t)
```

Comment:
Please order interfaces alphabetically and note sysnetwork, authconfig, miscfiles are modules.

Reply:
I ordered kernel modules and then system modules. It is ok now?

Reply:
ok, thank you